DOC-847: Updated docs for table. (#2025)
tinymce/tinymce-docs: release-notes/release-notes59.md

For information on the `toolbar_sticky_offset` option, see: [User interface options - toolbar_sticky_offset]({{site.baseurl}}/configure/editor-appearance/#toolbar_sticky_offset).

### New table commands

A series of new commands have been added to the table plugin to improve the usability of the plugin.
- The new command `mceTableCellToggleClass` allows the user to add or remove a class to selected cells .
- The new command `mceTableToggleClass` allows the user to add or remove a class to selected tables.
- The new command `mceTableToggleCaption` allows the user to add or remove a caption to selected tables.

> ### New table buttons

Reviewer: not resolved.

> Several new buttons and menu items have been added to further improve the user experience when manipulating tables.
> - The new button and menu item `tableclass` which allows a user to add or remove a class from selected tables. Available classes are defined by the [`table_class_list` option]({{site.baseurl}}/plugins/opensource/table/#table_class_list).
> - The new button and menu item `tablecellclass` which allows a user to add or remove a class from selected cells. Available classes are defined by the [`table_cell_class_list` option]({{site.baseurl}}/plugins/opensource/table/#table_cell_class_list).
> - The new button and menu item `tablecellvalign` which allows a user to set the alignment for the content of selected cells.
> - The new button and menu item `tablecellborderwidth` which allows quick access to setting the width of the border of table cells. The available widths are set with the [`table_border_widths` option]({{site.baseurl}}/plugins/opensource/table/#table_border_widths).
> - The new button and menu item `tablecellborderstyle` which allows quick access to setting the style of the border of table cells. The available styles are set with the [`table_border_styles` option]({{site.baseurl}}/plugins/opensource/table/#table_border_styles).
> - The new button and menu item `tablecaption` which allows toggling the caption of selected tables.
> - The new button and menu item `tablecellbackgroundcolor` allows the user to select the background color of selected cells. The available colors are set by the [`table_cell_background_color_map` option]({{site.baseurl}}/plugins/opensource/table/#table_cell_background_color_map) or [`color_map` option]({{site.baseurl}}/plugins/opensource/table/#color_map).
> - The new button and menu item `tablecellbordercolor` allows the user to select the border color of selected cells. The available colors are set by the [`table_cell_border_color_map` option]({{site.baseurl}}/plugins/opensource/table/#table_cell_border_color_map) or [`color_map` option]({{site.baseurl}}/plugins/opensource/table/#color_map).
> - The new button and menu item `tablerowheader` allows the user to toggle selected rows into headers.
> - The new button and menu item `tablecolheader` allows the user to toggle selected columns into headers.

Reviewer: ```suggestion
{{site.productname}} 5.9 adds new buttons and menu items for the Table plugin to improve the user experience when working with tables. The toolbar buttons and menu item use the same name (identifier). These toolbar buttons and menu items can be added to contextual menus (right-click menus) and contextual toolbars such as the table toolbar. The following toolbar buttons and menu items have been added:

`tableclass`
: Allows the user to add or remove a class from selected tables. Available classes are defined by the [`table_class_list` option]({{site.baseurl}}/plugins/opensource/table/#table_class_list).

`tablecellclass`
: Allows the user to add or remove a class from selected cells. Available classes are defined by the [`table_cell_class_list` option]({{site.baseurl}}/plugins/opensource/table/#table_cell_class_list).

`tablecellvalign`
: Allows the user to set the vertical alignment for the content of selected cells.

`tablecellborderwidth`
: Allows the user to set the border width on selected table cells. The available widths are set with the [`table_border_widths` option]({{site.baseurl}}/plugins/opensource/table/#table_border_widths).

`tablecellborderstyle`
: Allows the user to set the border style on selected table cells. The available styles are set with the [`table_border_styles` option]({{site.baseurl}}/plugins/opensource/table/#table_border_styles).

`tablecaption`
: Toggles the caption on selected tables.

`tablecellbackgroundcolor`
: Allows the user to change the background color of selected cells. The available colors are set by the [`table_cell_background_color_map` option]({{site.baseurl}}/plugins/opensource/table/#table_cell_background_color_map) or [`color_map` option]({{site.baseurl}}/plugins/opensource/table/#color_map).

`tablecellbordercolor`
: Allows the user to change the border color of selected cells. The available colors are set by the [`table_cell_border_color_map` option]({{site.baseurl}}/plugins/opensource/table/#table_cell_border_color_map) or [`color_map` option]({{site.baseurl}}/plugins/opensource/table/#color_map).

`tablerowheader`
: Allows the user to toggle selected rows into headers.

`tablecolheader`
: Allows the user to toggle selected columns into headers.
```

Reviewer: please apply James's suggestions to the above.

`tablecellbordercolor`
: Allows the user to change the border color of selected cells. The available colors are set by the [`table_cell_border_color_map` option]({{site.baseurl}}/plugins/opensource/table/#table_cell_border_color_map) or [`color_map` option]({{site.baseurl}}/plugins/opensource/table/#color_map).

`tablerowheader`
: Allows the user to toggle selected rows into headers.

`tablecolheader`
: Allows the user to toggle selected columns into headers.

For information on these buttons, see: [Available toolbar buttons]({{site.baseurl}}/advanced/available-toolbar-buttons/#tableplugin)
For information on these menu items, see: [Available menu items]({{site.baseurl}}/advanced/available-menu-items/#tableplugin)

#### New table options

{{site.productname}} 5.9 adds new options for the Table plugin to improve the user experience when working with tables. These new options are optional, and allows further customization of user experience.

`table_cell_background_color_map`

> : Allows setting a specific set of colors to be used by the `tablecellbackgroundcolor` toolbar button and menu item, overriding the defaults and the colors provided by the [`color_map` option]({{site.baseurl}}/plugins/opensource/table/#color_map). For information on this option, see: [Table options - table_cell_background_color_map]({{site.baseurl}}/plugins/opensource/table/#table_cell_background_color_map)

Reviewer 2: for added clarity 

Allows setting a specific set of background colors to be used by the

Reviewer: ```suggestion
: Allows setting a specific set of background colors to be used by the `tablecellbackgroundcolor` toolbar button and menu item, overriding the defaults and the colors provided by the [`color_map` option]({{site.baseurl}}/plugins/opensource/table/#color_map). For information on this option, see: [Table options - table_cell_background_color_map]({{site.baseurl}}/plugins/opensource/table/#table_cell_background_color_map)
```

`table_cell_border_color_map`

> : Allows setting a specific set of colors to be used by the `tablecellbordercolor` toolbar button and menu item, overriding the defaults and the colors provided by the [`color_map` option]({{site.baseurl}}/plugins/opensource/table/#color_map). For information on this option, see: [Table options - table_cell_border_color_map]({{site.baseurl}}/plugins/opensource/table/#table_cell_border_color_map)

Reviewer 2: for added clarity

Allows setting a specific set of border colors

Reviewer: ```suggestion
: Allows setting a specific set of border colors to be used by the `tablecellbordercolor` toolbar button and menu item, overriding the defaults and the colors provided by the [`color_map` option]({{site.baseurl}}/plugins/opensource/table/#color_map). For information on this option, see: [Table options - table_cell_border_color_map]({{site.baseurl}}/plugins/opensource/table/#table_cell_border_color_map)
```

`table_border_widths`
: Allows setting a specific set of widths to be used by the `tablecellborderwidth` toolbar button and menu item. For information on this option, see: [Table options - table_border_widths]({{site.baseurl}}/plugins/opensource/table/#table_border_widths)

`table_border_styles`

> : Allows setting a specific set of widths to be used by the `tablecellborderstyle` toolbar button and menu item. For information on this option, see: [Table options - table_border_styles]({{site.baseurl}}/plugins/opensource/table/#table_border_styles).

Reviewer 2: correction: Allows setting a specific set of widths to be used by the

Allows setting a specific HTML border style to be used by the

Reviewer: ```suggestion
: Allows setting a specific set of HTML border style to be used by the `tablecellborderstyle` toolbar button and menu item. For information on this option, see: [Table options - table_border_styles]({{site.baseurl}}/plugins/opensource/table/#table_border_styles).
```

> : Allows setting a specific set of HTML border style to be used by the `tablecellborderstyle` toolbar button and menu item. For information on this option, see: [Table options - table_border_styles]({{site.baseurl}}/plugins/opensource/table/#table_border_styles).

Reviewer 3: ```suggestion
: Allows setting a specific set of HTML border styles to be used by the `tablecellborderstyle` toolbar button and menu item. For information on this option, see: [Table options - table_border_styles]({{site.baseurl}}/plugins/opensource/table/#table_border_styles).
```

- The `content_langs` option, see: [Content appearance options - `content_langs`]({{site.baseurl}}/configure/localization/#content_langs).
- The `Lang` editor command, see: [Commands available for {{site.productname}} - Core Editor commands]({{site.baseurl}}/advanced/editor-command-identifiers/#coreeditorcommands).

### New table commands

A series of new commands have been added to the table plugin to improve the usability of the plugin.
- The new command `mceTableCellToggleClass` allows the user to add or remove a class to selected cells .

> - The new command `mceTableToggleClass` allows the user to add or remove a class to selected tables.

Reviewer 4: ```suggestion
- The new command `mceTableToggleClass` allows the user to add or remove a class to the selected table.
```

Reviewer 4: For some reason Github isn't letting me comment on the line below but it needs fixing too

- The new command `mceTableToggleCaption` allows the user to add or remove a caption to selected tables.

### New table buttons

Several new buttons and menu items have been added to further improve the user experience when manipulating tables.

> - The new button and menu item `tableclass` which allows a user to add or remove a class from selected tables. Available classes are defined by the [`table_class_list` option]({{site.baseurl}}/plugins/opensource/table/#table_class_list).

Reviewer 4: ```suggestion
- The new button and menu item `tableclass` which allows a user to add or remove a class from the selected table. Available classes are defined by the [`table_class_list` option]({{site.baseurl}}/plugins/opensource/table/#table_class_list).
```

- The new button and menu item `tablecellclass` which allows a user to add or remove a class from selected cells. Available classes are defined by the [`table_cell_class_list` option]({{site.baseurl}}/plugins/opensource/table/#table_cell_class_list).
- The new button and menu item `tablecellvalign` which allows a user to set the alignment for the content of selected cells.
- The new button and menu item `tablecellborderwidth` which allows quick access to setting the width of the border of table cells. The available widths are set with the [`table_border_widths` option]({{site.baseurl}}/plugins/opensource/table/#table_border_widths).
- The new button and menu item `tablecellborderstyle` which allows quick access to setting the style of the border of table cells. The available styles are set with the [`table_border_styles` option]({{site.baseurl}}/plugins/opensource/table/#table_border_styles).

> - The new button and menu item `tablecaption` which allows toggling the caption of selected tables.

Reviewer 4: ```suggestion
- The new button and menu item `tablecaption` which allows toggling the caption of the selected table.
```

- The new button and menu item `tablecellbackgroundcolor` allows the user to select the background color of selected cells. The available colors are set by the [`table_cell_background_color_map` option]({{site.baseurl}}/plugins/opensource/table/#table_cell_background_color_map) or [`color_map` option]({{site.baseurl}}/plugins/opensource/table/#color_map).
- The new button and menu item `tablecellbordercolor` allows the user to select the border color of selected cells. The available colors are set by the [`table_cell_border_color_map` option]({{site.baseurl}}/plugins/opensource/table/#table_cell_border_color_map) or [`color_map` option]({{site.baseurl}}/plugins/opensource/table/#color_map).
- The new button and menu item `tablerowheader` allows the user to toggle selected rows into headers.
- The new button and menu item `tablecolheader` allows the user to toggle selected columns into headers.
### New Table plugin features

#### New table commands

The following commands have been added to the table plugin to improve the usability:
`mceTableCellToggleClass`
: Allows the user to add or remove a class on selected cells.
`mceTableToggleClass`

> : Allows the user to add or remove a class on selected tables.

Reviewer 4: ```suggestion
: Allows the user to add or remove a class on the selected table.
```

`mceTableToggleCaption`

> : Allows the user to add or remove a caption on selected tables.

Reviewer 4: ```suggestion
: Allows the user to add or remove a caption on the selected table.
```

For information on these commands, see: [Table - Commands]({{site.baseurl}}/plugins/opensource/table/#commands)

#### New table toolbar buttons and menu items

{{site.productname}} 5.9 adds new buttons and menu items for the Table plugin to improve the user experience when working with tables. The toolbar buttons and menu item use the same name (identifier). These toolbar buttons and menu items can be added to contextual menus (right-click menus) and contextual toolbars such as the table toolbar. The following toolbar buttons and menu items have been added:
`tableclass`

> : Allows the user to add or remove a class from selected tables. Available classes are defined by the [`table_class_list` option]({{site.baseurl}}/plugins/opensource/table/#table_class_list).

Reviewer 4: ```suggestion
: Allows the user to add or remove a class from the selected table. Available classes are defined by the [`table_class_list` option]({{site.baseurl}}/plugins/opensource/table/#table_class_list).
```

`tablecellclass`
: Allows the user to add or remove a class from selected cells. Available classes are defined by the [`table_cell_class_list` option]({{site.baseurl}}/plugins/opensource/table/#table_cell_class_list).
`tablecellvalign`
: Allows the user to set the vertical alignment for the content of selected cells.
`tablecellborderwidth`
: Allows the user to set the border width on selected table cells. The available widths are set with the [`table_border_widths` option]({{site.baseurl}}/plugins/opensource/table/#table_border_widths).
`tablecellborderstyle`
: Allows the user to set the border style on selected table cells. The available styles are set with the [`table_border_styles` option]({{site.baseurl}}/plugins/opensource/table/#table_border_styles).
`tablecaption`

> : Toggles the caption on selected tables.

Reviewer 4: ```suggestion
: Toggles the caption on the selected table.
```

`tablecellborderstyle`
: Allows the user to set the border style on selected table cells. The available styles are set with the [`table_border_styles` option]({{site.baseurl}}/plugins/opensource/table/#table_border_styles).
`tablecaption`
: Toggles the caption on selected tables.
`tablecellbackgroundcolor`
: Allows the user to change the background color of selected cells. The available colors are set by the [`table_cell_background_color_map` option]({{site.baseurl}}/plugins/opensource/table/#table_cell_background_color_map) or [`color_map` option]({{site.baseurl}}/plugins/opensource/table/#color_map).
`tablecellbordercolor`
: Allows the user to change the border color of selected cells. The available colors are set by the [`table_cell_border_color_map` option]({{site.baseurl}}/plugins/opensource/table/#table_cell_border_color_map) or [`color_map` option]({{site.baseurl}}/plugins/opensource/table/#color_map).
`tablerowheader`
: Allows the user to toggle selected rows into headers.
`tablecolheader`
: Allows the user to toggle selected columns into headers.

### New Table plugin features

> #### New table commands

Reviewer 3: This information is duplicated. It looks like the sections above should not exist.

- [ ] Fix duplicate content
